### PR TITLE
bld2repo: fix buildrequires based on MBS data

### PR DESCRIPTION
**Pull request by mkulik-rh**
RPMs provided by `get_buildrequire_pkgs_from_build` to download so additionally filtered using MBS data in `filter_buildrequire_pkgs` to eliminate downloading unnecessary dependencies based on Buildorder.

New function: filter_buildrequire_pkgs
 - Filters output provided by get_buildrequire_pkgs_from_build
New function: get_koji_build_info
 - provides build information for given build_id

Old tests fixed due to changes and new test added to cover filtering.

### CODE REVIEW

**Comment by FrostyX** [comment]
@mcurlej, can you please take a look?
I don't use this tool.

**Comment by kdudka** [comment]
How can I easily install `bld2repo` from this pull request on my system?

**Comment by mkulik-rh** [comment]
Here is one way, bld2repo is only one part of many tools in modulemd-tools package:
```
#!/bin/bash
dnf install -y git tito dnf-plugins-core
dnf -y builddep modulemd-tools
git clone https://github.com/mkulik-rh/modulemd-tools.git
git checkout MODULAR-179-v3
cd modulemd-tools
tito build --rpm --install --test --offline
```

I'm not sure if this part is necessary and maybe it can be removed:

https://github.com/mkulik-rh/modulemd-tools/blob/8ab81f3c62a36f94ae0c1973401b7e139df1d1e8/modulemd_tools/bld2repo/__init__.py#L120-121

**Comment by kdudka** [comment]
Thanks for the hint!  I was able to install it using tito.

**Comment by mcurlej** [comment]
LGTM. Just add tests to validate the exceptions in the new functions (missing id and missing modulemd). Btw it seems the CI died.

**Comment by mkulik-rh** [comment]
@mcurlej Can you quickly re-check it ?

**Comment by mcurlej** [comment]
LGTM! two things. The typo i found earlier is still there. The second please add some info about the the new mbs parameter into this file https://github.com/rpm-software-management/modulemd-tools/blob/master/README-DETAILED.md (i am not sure why there are 2 README documents.)

File: modulemd_tools/bld2repo/__init__.py [hunk 1]
```diff
@@ -51,6 +70,72 @@ def get_buildrequire_pkgs_from_build(build_id, session, config):
     return pkgs
 
 
+def filter_buildrequire_pkgs(build, pkgs, config):
+    """
+    Filter RPMs based on BuildOrder from Mobile Build Service.
```
[the remark this answers is not in the document]
Comment (mkulik-rh) [reply modulemd_tools/bld2repo/__init__.py]: Fixed

**Pull request merged**